ops/testing.py

```diff
@@ -1574,8 +1575,41 @@ def remove_path(self, path: str, *, recursive: bool = False):
     def exec(self, command, **kwargs):
         raise NotImplementedError(self.exec)
 
-    def send_signal(self, sig: typing.Union[int, str], services: typing.List[str]):
-        raise NotImplementedError(self.send_signal)
+    def send_signal(self, sig: typing.Union[int, str], *service_names: str):
+        if not service_names:
+            raise TypeError('send_signal expected at least 1 service name, got 0')
+
+        # Convert signal to str
+        if isinstance(sig, int):
+            sig = signal.Signals(sig).name
+
+        # pebble first validates the service name, and then the signal name
+
+        def is_running(service: str) -> bool:
+            # Scalar helper function for the vectorized is_running().
+            return next(iter(self.get_services([service]))).is_running()
+
+        plan = self.get_plan()
+        for service in service_names:
+            if service not in plan.services or not is_running(service):
+                # conform with the real pebble api
+                raise pebble.APIError(
+                    body={}, code=500, status='Internal Server Error',
```

Comment:
In reality, `body` looks like this:
`{'type': 'error', 'status-code': 500, 'status': 'Internal Server Error', 'result': {'message': 'cannot send signal to "ss": service is not running'}}`
but seems like it is not needed for the harness..?

Reply:
OF is probably just watching for the pebble.APIError, without inspecting it.

It would be good code hygiene to refactor the error internals to match the production ones, though.